### DISCUSSION TITLE
ConvexHullShape Constructor That Takes Planes

### DISCUSSION THREAD

**Discussion by a-day-old-bagel** [jrouwe/JoltPhysics#1084] [Ideas]
Would it be possible to get a construction method for a ConvexHullShape that lets the user provide the planes directly rather than the points for the hull? I have a use case where it's easier to provide the planes than the points.

**Comment by jrouwe** [comment]
Unfortunately, the ConvexHullShape uses the vertices in its internal representation, so if I were to offer an interface to supply the planes then I would need to write something to calculate the vertices anyway.

A brute force approach would be:

```
for each plane p1
	for each 2nd plane p2
		for each 3rd plane p3
			Vec3 p
			Plane::sIntersectPlanes(p1, p2, p3, p)
			for all planes p4
				if p4.SignedDistance(p) > 0
					discard this vertex
			accept p as an input vertex (you may get some interior vertices but these will be discarded later)
```

I haven't done any investigation if there's a more performant approach to go from planes to vertices.

**Comment by a-day-old-bagel** [comment]
Ok no worries then. I hadn't realized the internal representation is primarily as points, and had thought it was equivalently treated as planes, given that the structure appears to store planes in mPlanes.

I'll just do the computation myself then, as before. Thanks for taking the time to respond.
